### PR TITLE
Highest/Lowest Scoring Home and Visitor Teams

### LINKED ISSUE #21: Build method worst_defense (return String)
Name of the team with the highest average number of goals allowed per game across all seasons.

### LINKED ISSUE #23: Build method highest_scoring_home_team (return String)
Name of the team with the highest average score per game across all seasons when they are home.

### LINKED ISSUE #24: Build method lowest_scoring_visitor (return String)
Name of the team with the lowest average score per game across all seasons when they are a visitor.

### LINKED ISSUE #25: Build method lowest_scoring_home_team (return String)
Name of the team with the lowest average score per game across all seasons when they are at home.

### PR DESCRIPTION
**Pull request by linda-le1**
## What functionality does this accomplish?
closes #21 
closes #23 
closes #24 
closes #25 

**Description:**
Included the following methods with passing tests:

highest_scoring_visitor
highest_scoring_home_team
lowest_scoring_visitor
lowest_scoring_home_team

When run in the runner.rb file and checked with pry, these methods do return what is expected per RSPEC. For some reason, RSPEC itself is not passing these tests when executed though.

## What did you struggle on to complete?

Getting all of the helper methods built, but once that was done, the rest were easy to write up. For sure could use a module here or some refactoring.

### Test Coverage Percentage: x%
- [ ] No Tests have been changed
- [X] New Tests have been implemented
- [ ] Some Tests have been changed
- [ ] All of the Tests have been changed(Please describe what in the world happened):

## Checklist:
- [X] My code has no unused/commented out code
- [X] I have reviewed my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have fully tested my code
- [X] I have partially tested my code (please explain why): Using pry, these methods get the correct values as noted in the RSPEC and the methods are named correctly, but RSPEC is not passing them.

## Helpful Resources:
* Resource link AND small description of info gathered/learned

### CODE REVIEW

File: lib/team_collection.rb [hunk 1]
```diff
@@ -17,6 +17,13 @@ def total_teams
     @teams.length
   end
 
+  def find_team_name_by_id(team_id)
```
Comment (jfangonilo) [inline lib/team_collection.rb]: Please refactor to use recently pushed version of find_by_id

**Review by jfangonilo — CHANGES_REQUESTED** [review]
Looks good as long as it passes the spec harness, just need to refactor usage of find_by_id

File: lib/game_collection.rb [hunk 2]
```diff
@@ -67,4 +67,105 @@ def find_percentage_of_ties
...
+      end
+    end
+    number_of_home_games
+  end
+
+  def find_number_of_away_games
+    number_of_away_games = Hash.new()
+    @games.each do |game|
+      away_team_id = game.away_team_id.to_i
+      if number_of_away_games.has_key?(away_team_id)
```
Comment (jobannon) [inline lib/game_collection.rb:89]: Like this .has Key ... super nice

**Review by jfangonilo — APPROVED** [review]

**Comment by jfangonilo** [comment]
Looks good, change to take team_id as string was necessary also.

**Pull request merged**